### PR TITLE
Backport missing PRs from Bitcoin 0.14 - Part 4

### PR DESCRIPTION
**Pull request by codablock**
Continuation of https://github.com/dashpay/dash/pull/1770

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# Copyright (c) 2013-2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
```
Comment (UdjinM6) [inline Makefile.am]: Most `Makefile`s (besides probably `src/Makefile.leveldb.include`) are modified by us already, so we should probably add a `// Copyright (c) 2014-2017 The Dash Core developers` line as well (note: this comment applies to files modified in a26bad09b6a31220583164f78efbbb1e5de2525e).

File: INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Building Bitcoin
```
Comment (UdjinM6) [inline INSTALL.md]: s/Bitcoin/Dash Core/

File: INSTALL.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+================
+
+See doc/build-*.md for instructions on building the various
+elements of the Bitcoin Core reference implementation of Bitcoin.
```
Comment (UdjinM6) [inline INSTALL.md]: s/Bitcoin/Dash/

File: doc/build-unix.md [hunk 4]
```diff
@@ -308,3 +308,35 @@ To build executables for ARM:
...
+(`libressl` instead of `openssl` will also work)
+
+For the wallet (optional):
+
+    pkg install db5
+
+This will give a warning "configure: WARNING: Found Berkeley DB other
+than 4.8; wallets opened by this build will not be portable!", but as FreeBSD never
+had a binary release, this may not matter. If backwards compatibility
+with 4.8-built Bitcoin Core is needed follow the steps under "Berkeley DB" above.
```
Comment (UdjinM6) [inline doc/build-unix.md]: s/Bitcoin/Dash/

File: src/random.h [hunk 5]
```diff
@@ -28,26 +28,23 @@ uint256 GetRandHash();
...
 
 /**
- * Seed insecure_rand using the random pool.
- * @param Deterministic Use a deterministic seed
+ * Fast randomness source. This is seeded once with secure random data, but
+ * is completely deterministic and insecure after that.
+ * This class is not thread-safe.
  */
-void seed_insecure_rand(bool fDeterministic = false);
+class FastRandomContext {
```
Comment (UdjinM6) [inline src/random.h]: `FastRandomContext` is basically a clone of `InsecureRand` which was implemented in Dash earlier. I think it makes sense to s/InsecureRand/FastRandomContext/ elsewhere in code and drop `InsecureRand` in `random.h/cpp` now.

File: src/dashd.cpp [hunk 6]
```diff
@@ -142,29 +143,21 @@ bool AppInit(int argc, char* argv[])
...
-            {
-                return true;
-            }
-            // Child process falls through to rest of initialization
-
-            pid_t sid = setsid();
-            if (sid < 0)
-                fprintf(stderr, "Error: setsid() returned %d errno %d\n", sid, errno);
+#else
+            fprintf(stderr, "Error: -daemon is not supported on this operating system\n");
```
Comment (gladcow) [inline src/dashd.cpp]: Now I'm getting this error on Ubuntu 17. I'm trying to understand details now,  but it looks like a problem

File: src/test/data/bitcoin-util-test.json [hunk 7]
```diff
@@ -1,37 +1,73 @@
 [
   { "exec": "././dash-tx",
```
Comment (UdjinM6) [inline src/test/data/bitcoin-util-test.json]: s/`"././dash-tx"`/`"./dash-tx"`/

File: src/dashd.cpp (hunk 6 above)
Comment (UdjinM6) [reply src/dashd.cpp]: Make sure to `make clean && ./autogen.sh`

**Comment by codablock** [comment]
@UdjinM6 Regarding the missing change in doc/release-process.md, I've omitted quite a lot in release-process.md as it has diverged much from Bitcoin (partly due to custom changes, partly due to out of order backporting). I have an item on my TODO to go through the release-process.md when I reach Bitcoin 0.15 to compare and finally add all missing stuff.

Working on the other stuff now.

File: src/dashd.cpp (hunk 6 above)
Comment (gladcow) [reply src/dashd.cpp]: Yes, I'm sorry, it's my error,  I've missed autogen.sh.

**Comment by codablock** [comment]
Fixed review comments

**Pull request merged**